torch/distributed/rpc.py

```diff
@@ -62,10 +73,19 @@ def _init_rpc(name, backend=RpcBackend.PROCESS_GROUP):
         raise RuntimeError("RPC is already initialized")
 
     if backend == RpcBackend.PROCESS_GROUP:
-        from .distributed_c10d import _get_default_group
+        from .distributed_c10d import _get_default_group, init_process_group
+
+        init_process_group(
+            backend="gloo", rank=rank, world_size=world_size, store=store
+        )
         group = _get_default_group()
         # TODO: add try-except and destroy _agent in all processes if any fails.
         _agent = ProcessGroupAgent(name, group)
+    elif is_backend_registered(backend):
+        _agent = registered_init_rpc(backend,
+                                     worker_id=rank,
+                                     worker_names=["worker{}".format(i) for i in range(world_size)],
```

Comment:
Let's add a `TODO` to collect worker names here.

Reply:
solved with URL where one can pass in a store or something to identify the workers.